scripts/localStorage.js

```diff
@@ -0,0 +1,98 @@
+/*
+Minecraft Pixel Art Maker
+© gd-codes 2024
+https://gd-codes.github.io/mc-pixelart-maker/
+*/
+
+
+function saveFormData(uid, {
+  PictureDataForUid: {
+    originalFileName,
+    originalImage,
+    resizedImage,
+    originalWasResized
+  },
+  fnName,
+  configuration
+}) {
+
+  // We prefer to store the original, but if it exceeds the threshold and the resized is smaller than the original,
+  // we store the resized instead.
+  // In that case, we store the resized even if it exceeds the threshold.
+  let originalTooLarge = 1000000 < originalImage.length &&
```

Comment:
Note: This threshold is based on the base64 encoded image. The current limit of 1 MB allows saving an original image of 730 KB file size.
I picked a limit of 1MB basically arbitrarily after reading that local storage is usually limited to 5-10 MB.
We could also omit or increase the threshold to support larger images at the cost of storing less images overall.